### DISCUSSION TITLE
QueryError(PostgresError { code: "26000", message: "prepared statement \"s0\" does not exist"

### DISCUSSION THREAD

**Discussion by Jhonypd** [prisma/prisma#22837] [Q&A] [answered]
### Question

I'm using this instance for the prism:
```
import { PrismaClient } from "@prisma/client";

declare global {
  var cachedPrisma: PrismaClient;
}

let prisma: PrismaClient;

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient();
} else {
  if (!global.cachedPrisma) {
    global.cachedPrisma = new PrismaClient();
  }
  prisma = global.cachedPrisma;
}

export const prismaClient = prisma;
```

when I call:

```
const vendedores = await prismaClient.usuario.findMany({
    distinct: ["nome"],
  });
```
  
  I start getting this error:

```  
  Invalid `prisma.usuario.findMany()` invocation:


Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(PostgresError { code: "26000", message: "prepared statement \"s1\" does not exist", severity: "ERROR", detail: None, column: None, hint: None }), transient: false })       
    at async Filters (components/filters.tsx:15:24)
digest: "985112066"
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by janpio** [comment]
Where do you host your PostgreSQL database?
You are probably looking for https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/databases-connections/pgbouncer

**Reply by dookanooka** [reply]
I get a similar error after switching Supabase connection URL from pgbouncer to their AWS connection pooling
```
Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(PostgresError { code: "26000", message: "prepared statement \"s0\" does not exist", severity: "ERROR", detail: None, column: None, hint: None }), transient: false })
```

see [https://github.com/orgs/supabase/discussions/17817](https://github.com/orgs/supabase/discussions/17817)

**Reply by Jhonypd** [reply, marked as answer]
Someone told me to modify the connection string from:
`DATABASE_URL="postgres://postgres.cbdhekfhcknrsdxfplka:password@aws-0-us-west-1.pooler.supabase.com:6543/postgres?pgbouncer=true"`

for:
`DATABASE_URL="postgres://postgres.cbdhekfhcknrsdxfplka:password@aws-0-us-west-1.pooler.supabase.com"`

It worked on mine, I don't know if it will cause any problems in the future.

**Reply by dookanooka** [reply]
Leaving out the :6543/postgres worked for me. I'd already removed the pgbouncer and gone to the connection pool URL as we've gone past the 26th Jan, thanks Jhonypd

**Reply by narcis-fv** [reply]
This worked for me as well with the new supabase string. Thank you!

**Reply by yusufrachmad** [reply]
this worked for me also. Big thanks Jhonypd

**Reply by mohamm-ad** [reply]
also worked, thank you!

**Reply by NEARworld** [reply]
perfectly worked thansk!

**Comment by Jhonypd** [comment]
Even though I make the settings according to the documentation, I keep getting errors:

```
Data source "db": PostgreSQL database "postgres", schema "public" at "db.cbdhekfhcknrsdxfplka.supabase.co:5432"

Error: P1001: Unable to access database server at `db.cbdhekfhcknrsdxfplka.supabase.co`:`5432`

Please make sure your database server is running at `db.cbdhekfhcknrsdxfplka.supabase.co`:`5432`.
```

my connection strings:

`DATABASE_URL="postgres://postgres.cbdhekfhcknrsdxfplka:password@aws-0-us-west-1.pooler.supabase.com:6543/postgres?pgbouncer=true"
`
`DIRECT_URL="postgresql://postgres:password@db.cbdhekfhcknrsdxfplka.supabase.co:5432/postgres"`

**Reply by BrightonMboya** [reply]
you might want to hide ur project id or sth

**Comment by BrightonMboya** [comment]
@Jhonypd Supabase updated their prisma guide here https://supabase.com/partners/integrations/prisma.

You have to append `?pgbouncer=true&connection_limit=1` after your connection strings. The `pgbouncer=true` is to prevent prisma from generating prepared statements and the other one is required if you're using Prisma in a serverless env

**Comment by ezeparziale** [comment]
Nextjs + prisma + supabase

DATABASE_URL="Transaction connection pooler string +  ?pgbouncer=true&connection_limit=1"
DIRECT_URL="Session connection pooler string"
